### PR TITLE
Feature/139 govuk react jsx components

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

- Replace temporary govuk components replicas with components from govuk-react-jsx library.
- Fix broken tests.
- Significant refactor of tests to improve code readability.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have run all unit tests
- [X] I have manually tested the application and related features

# Checklist:

- [X] My changes do not introduce any new linting errors 
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: designer/client/conditions/select-conditions.js [hunk 1]
```diff
@@ -79,28 +78,52 @@ class SelectConditions extends React.Component {
 
   render () {
     const { selectedCondition, inline } = this.state
-    const { data, hints } = this.props
+    const { data, hints = [] } = this.props
```
Comment (jenbutongit) [inline designer/client/conditions/select-conditions.js:81]: I discovered default destructure syntax recently.. very neat :D

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: designer/client/conditions/select-conditions.js [hunk 2]
```diff
@@ -79,28 +78,52 @@ class SelectConditions extends React.Component {
...
                 'Select a condition',
                 selectedCondition ?? '',
-                this.props.data.conditions.map((it, index) => (new SelectOption(it.displayName, it.name))),
+                this.props.data.conditions.map((it, index) => (new SelectOption(it.displayName, it.name))), // (text: string, value: string)
                 this.onChangeConditionSelection,
                 new SelectInputOptions(false, true, undefined, 'select-condition')
-              )
-            }
+                                      (required: boolean, includeBlankOption: boolean, hints: ?Array<string>, groupId: ?string)
+              )} */}
```
[the remark this answers is not in the document]
Comment (bertrandmc) [reply designer/client/conditions/select-conditions.js]: Ooops, cleaned. Thanks.

File: .yarn/build-state.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Warning: This file is automatically generated. Removing it is fine, but will
```
Comment (jenbutongit) [inline .yarn/build-state.yml:1]: delete please

**Review by jenbutongit — CHANGES_REQUESTED** [review]

**Comment by jenbutongit** [comment]
lgtm apart from build-state. could you delete and add to gitignore please

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**